### PR TITLE
Add Device View

### PR DESCRIPTION
**Pull request by tomjackman**
Add a Device View. Included detections to check:

1. If the App is running in an emulator
2. If there is an Android debugger attached
3. If the device that the app is running on is rooted
4. If there are any hooking frameworks installed on the device
5. If the user has setup a device lock screen (Android M or higher, otherwise don't perform/show in the UI)

Also added a small change to the pinning error checker.

The app was tested in an emulator with a debugger attached, with no lock screen setup and the xposed hooking framework installed on the device.
detections-final

The root detection functionality was tested on a separate sample application on an API Level 17 device and detected root successfully.
root-detected

### CODE REVIEW

File: app/src/main/java/com/feedhenry/securenativeandroidtemplate/features/device/DeviceFragment.java [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+        // Inflate the layout for this fragment
+        view = inflater.inflate(R.layout.fragment_device, container, false);
+        ButterKnife.bind(this, view);
+
+        // perform detections
+        detectRoot();
+        detectDeviceLock();
+        detectEmulator();
+        debuggerDetected();
+        detectHookingFramework();
```
Comment (wei-lee) [inline app/src/main/java/com/feedhenry/securenativeandroidtemplate/features/device/DeviceFragment.java]: what is the hooking framework?

**Comment by wei-lee** [comment]
👍  looks good!

**Comment by tomjackman** [comment]
- added Device Trust score percentage
- added check to see if Developer mode is enabled.
- added check to see if the device is running the latest OS version.
- added check to see if the device is using device level encryption

File: app/src/main/java/com/feedhenry/securenativeandroidtemplate/features/device/DeviceFragment.java (hunk 1 above)
Comment (tomjackman) [reply app/src/main/java/com/feedhenry/securenativeandroidtemplate/features/device/DeviceFragment.java]: Sorry, missed this comment. There are some frameworks that allow you to intercept function calls and replace them with your own - this is where the "hooking" name comes from. Mainly [Xposed](http://repo.xposed.info/module/de.robv.android.xposed.installer) and [Cydia Substrate](http://www.cydiasubstrate.com)

**Comment by tomjackman** [comment]
@wei-lee Quick review? I just added 3 extra checks since the last time.

File: app/src/main/java/com/feedhenry/securenativeandroidtemplate/features/device/DeviceFragment.java [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+    }
+
+    public void detectDeveloperOptions() {
+        totalTests++;
+        int devOptionsEnabled = Settings.Secure.getInt(context.getContentResolver(),
+                Settings.Global.DEVELOPMENT_SETTINGS_ENABLED ,
+                0);
+
+        if (devOptionsEnabled > 0)
+        {
```
Comment (wei-lee) [inline app/src/main/java/com/feedhenry/securenativeandroidtemplate/features/device/DeviceFragment.java]: formatting

**Comment by wei-lee** [comment]
👍

**Pull request merged**